README.md

```diff
@@ -24,10 +24,11 @@ below for explanation of `emit_*` calls):
 
     require 'parser/current'
     # opt-in to most recent AST format:
-    Parser::Builders::Default.emit_lambda   = true
-    Parser::Builders::Default.emit_procarg0 = true
-    Parser::Builders::Default.emit_encoding = true
-    Parser::Builders::Default.emit_index    = true
+    Parser::Builders::Default.emit_lambda              = true
```

Comment:
Might be a good idea to add some comments explaining what those options do.

Reply:
There *is* documentation right above explaining what these options do.

Reply:
The documentation in the `backwards compatibility` section is extremely generic and doesn't mention any of the options specifically. I'm pretty sure most newcomers would have try some trouble trying to figure out what some of the more weird options mean.

Reply:
The fact that we even *have* these options is a consequence of our errors in judgement earlier. They should never be turned off explicitly.

Reply:
Fair enough. :-) This can also be made clearer in the docs, but I guess it's not a big deal.

Reply:
> They should never be turned off explicitly, and in particular newcomers shouldn't set them to any other value.

That is the reason I propose to expose a (or the) `#modernize` method and document it. It removes even needing to know the details of what modernize means.

Reply:
> That is the reason I propose to expose a (or the) `#modernize` method and document it.

That's impossible. Ever changing what that method does would break backwards compatibility, thus defeating the purpose of these variables in first place.

Reply:
> That's impossible. Ever changing what that method does would break backwards compatibility, thus defeating the purpose of having it in first place.

True. The method itself would have to be versioned, defeating the propose.
For my use cases I'll probably detect "malformed from not modernized" ASTs and reject them early.

Okay, so can we merge the PR as is? I think its an incremental improvement over the current state.